ianalyzer_readers/readers/rdf.py

```diff
@@ -0,0 +1,45 @@
+from typing import Iterable
+
+from rdflib import BNode, Graph
+
+from .core import Reader, Document, Source
+
+
+class RDFReader(Reader):
+    '''
+    A base class for Readers of Resource Description Framework files.
+    These could be in Turtle, JSON-LD, RDFXML or other formats,
+    see [rdflib parsers](https://rdflib.readthedocs.io/en/stable/plugin_parsers.html)
+    '''
+
+    def source2dicts(self, source: Source) -> Iterable[Document]:
+        '''
+        Given a RDF source file, returns an iterable of extracted documents.
+
+        Parameters:
+            source: the source file to extract. This is a string of the file path
+
+        Returns:
+            an iterable of document dictionaries. Each of these is a dictionary,
+                where the keys are names of this Reader's `fields`, and the values
+                are based on the extractor of each field.
+        '''
+        filename = source
+        g = Graph()
+        g.parse(filename)
+        document_subjects = self.document_subjects(g)
+        for subject in document_subjects:
+            yield self._document_from_subject(g, subject)
+
+    def document_subjects(self, graph: Graph) -> list:
```

Comment:
Small technical comment: it looks like this function could return any iterable, not just a list. The return types of rdflib query functions are often generators, so it's good to be clear that it's _not_ necessary to collect results in a list.